packages/flutter_tools/bin/tool_backend.dart

```diff
@@ -42,9 +48,12 @@ or
 ''');
     exit(1);
   }
-
-  final String flutterExecutable = path.join(
-    flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter');
+  String flutterExecutable;
+  if (Platform.isWindows) {
+    flutterExecutable = '$flutterRoot\\bin\\flutter.bat';
+  } else {
+    flutterExecutable = '$flutterRoot/bin/flutter';
+  }
```

Comment:
Maybe we should make a little `join_paths` helper function for this file that does what `path.join` does, so we can keep the simpler inline logic (and don't need to duplicate it more times if we need to add more paths)?

Reply:
Done